### PR TITLE
Defer offer confirmation screen should show next cycle name relative to application

### PR DESCRIPTION
**Pull request by steventux**
## Context

When they try to defer an offer (e.g. with pending conditions) that was made in the previous cycle (2020), the confirmation screen states that the application will be deferred to 2022-2023, which is the cycle after the one they want. This is because the rendered date is relative to the current cycle.

## Changes proposed in this pull request

- Adds a common parameterised method to render cycle names.
- Uses the method with the recruitment cycle from the application in the deferral confirmation view.

## Link to Trello card

https://trello.com/c/98QAgAQX/2962-defer-offer-confirmation-screen-shows-incorrect-cycle-name

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/models/recruitment_cycle.rb [hunk 1]
```diff
@@ -29,10 +29,14 @@ def self.years_visible_in_support
...
-    "#{current_year} to #{next_year}"
+    cycle_name
   end
 
   def self.next_cycle_name
-    "#{next_year} to #{next_year + 1}"
+    cycle_name(next_year)
+  end
+
+  def self.cycle_name(year = current_year)
```
[the remark this answers is not in the document]
Comment (mnacos) [reply app/models/recruitment_cycle.rb:39]: Is it worth having separate `recruitment_cycle_name` and `training_cycle_name`? For example, recruitment for a course with `recruitment_cycle_year` of 2021 happens in 2020-2021, but the actual training happens in 2021-2022. I realise the same `cycle_name` can be used for both cases, by adjusting the year passed, but forcing us dev to choose between `recruitment_cycle_name` and `training_cycle_name` would make the intention clearer in views/emails etc.

**Review by mnacos — APPROVED** [review]
I think this PR solves the immediate need (fixing the copy when deferring previous cycle applications) but there is another discussion to be had around the business rules of the deferrals feature. I've added an optional minor enhancement request for clarifying the `cycle_name` context.

File: app/models/recruitment_cycle.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (steventux) [reply app/models/recruitment_cycle.rb:31]: I agree the difference between recruitment and training cycles is not very obvious.
I've simplified the `RecruitmentCycle` model as the methods `current_cycle_name` and `next_cycle_name` are not used.
So the new `cycle_name` method seems to be the minimal we need for this particular bugfix, it also seems explicit enough that it concerns the recruitment cycle in this case given the model name.
I think you're right that we will eventually need to differentiate the two easily but it's not necessary for this bugfix.

**Review by despo — APPROVED** [review]
LGTM, nice little refactoring 👍🏼

**Pull request merged**